docs/src/api.md

```diff
@@ -0,0 +1,87 @@
+```@meta
+CurrentModule = StatsModels
+```
+
+# StatsModels.jl API
+
+## Formulae and terms
+
+```@docs
+@formula
+term
+coefnames
+model_cols
```

Comment:
We should probably avoid using underscores as much as possible for consistency with Julia Base and StatsBase (e.g. `coefnames` right above). Maybe that's not possible for all functions, but here `modelcols` sounds OK.